lib/usd/translators/meshWriter.cpp

```diff
@@ -160,44 +160,12 @@ PxrUsdTranslators_MeshWriter::writeMeshAttrs(
         return true;
     }
 
-    unsigned int numVertices = geomMesh.numVertices();
-    unsigned int numPolygons = geomMesh.numPolygons();
-
     // Set mesh attrs ==========
-    // Get points
-    // TODO: Use memcpy()
-    const float* mayaRawPoints = geomMesh.getRawPoints(&status);
-    VtArray<GfVec3f> points(numVertices);
-    for (unsigned int i = 0; i < numVertices; i++) {
-        unsigned int floatIndex = i*3;
-        points[i].Set(mayaRawPoints[floatIndex],
-                      mayaRawPoints[floatIndex+1],
-                      mayaRawPoints[floatIndex+2]);
-    }
+    // Write points
+    UsdMayaMeshUtil::writeVertexData(geomMesh, primSchema, usdTime, *_GetSparseValueWriter());
 
-    VtArray<GfVec3f> extent(2);
-    // Compute the extent using the raw points
-    UsdGeomPointBased::ComputeExtent(points, &extent);
-
-    _SetAttribute(primSchema.GetPointsAttr(), &points, usdTime);
-    _SetAttribute(primSchema.CreateExtentAttr(), &extent, usdTime);
-
-    // Get faceVertexIndices
-    unsigned int numFaceVertices = geomMesh.numFaceVertices(&status);
-    VtArray<int>     faceVertexCounts(numPolygons);
-    VtArray<int>     faceVertexIndices(numFaceVertices);
-    MIntArray mayaFaceVertexIndices; // used in loop below
-    unsigned int curFaceVertexIndex = 0;
-    for (unsigned int i = 0; i < numPolygons; i++) {
-        geomMesh.getPolygonVertices(i, mayaFaceVertexIndices);
-        faceVertexCounts[i] = mayaFaceVertexIndices.length();
-        for (unsigned int j=0; j < mayaFaceVertexIndices.length(); j++) {
-            faceVertexIndices[ curFaceVertexIndex ] = mayaFaceVertexIndices[j]; // push_back
-            curFaceVertexIndex++;
-        }
-    }
-    _SetAttribute(primSchema.GetFaceVertexCountsAttr(), &faceVertexCounts, usdTime);
-    _SetAttribute(primSchema.GetFaceVertexIndicesAttr(), &faceVertexIndices, usdTime);
+    // Write faceVertexIndices
+    UsdMayaMeshUtil::writeFaceVertexIndicesData(geomMesh, primSchema, usdTime, *_GetSparseValueWriter());
 
     // Read subdiv scheme tagging. If not set, we default to defaultMeshScheme
     // flag (this is specified by the job args but defaults to catmullClark).
```

Comment:
`writeInvisibleFacesData()` came from the block down below that handles holes. Shouldn't we be replacing that block with the new function?

Reply:
This is my bad here. As you probably have seen, this change happens in the next PR. I am not really sure why I did it this way originally and I admit it is confusing. :(